sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,252 @@
+            &column_type_ref.time_series_measure_value_column_info
+        {
+            let datapoint_list = datum
+                .time_series_value
+                .as_ref()
+                .ok_or("Time series value is None")?;
+            value.push_str("[");
+            value.push_str(&process_time_series_type(
+                &datapoint_list,
+                &time_series_measure_value_column_info,
+            )?);
+            value.push_str("]");
+        } else if let Some(array_column_info) = &column_type_ref.array_column_info {
+            let array_value = datum.array_value.as_ref().ok_or("Array value is None")?;
+            value.push_str("[");
+            value.push_str(&process_array_type(&array_value, array_column_info)?);
+            value.push_str("]");
+        } else if let Some(row_column_info) = &column_type_ref.row_column_info {
+            let row_value = datum.row_value.as_ref().ok_or("Row value is None")?;
+            value.push_str("[");
+            value.push_str(&process_row_type(&row_value.data, &row_column_info)?);
+            value.push_str("]");
+        } else {
+            panic!("Bad column type");
+        }
+
+        // Comma-seperated column values
+        if i != data.len() - 1 {
+            value.push_str(", ");
+        }
+    }
+    return Ok(value);
+}
+
+pub async fn run_query(
+    query: String,
+    client: &timestream_query::Client,
+    f: &std::fs::File,
+    max_rows: i32,
+) -> Result<(), String> {
+    let query_client = client.query().clone();
+
+    let mut query_result = query_client
+        .clone()
+        .max_rows(max_rows)
+        .query_string(&query)
+        .send()
+        .await;
+
+    let mut token: String;
+    let mut num_rows = 0;
+    loop {
+        match query_result {
+            Ok(query_success) => {
+                num_rows += query_success.rows.len();
+                if let Some(new_next_token) = query_success.next_token {
+                    // Set token to paginate through results
+                    token = new_next_token;
+                    query_result = query_client
+                        .clone()
+                        .max_rows(max_rows)
+                        .query_string(&query)
+                        .next_token(token)
+                        .send()
+                        .await;
+                } else {
+                    break;
+                }
+            }
+
+            Err(error) => {
+                let error_string = error.to_string();
+                let message = format!(
+                    "Error while querying the query {} : {}",
+                    &query, error_string
+                );
+                println!("{}", message);
+                let _ = write(f, message);
+                return Err(error_string);
+            }
+        }
+    }
+    let message = format!("Number of rows: {}", num_rows).to_string();
+    println!("{}", message);
+    let _ = write(f, message);
```

Comment:
We can return errors here instead.

```suggestion
    write(f, message)?;
```

Reply:
I have changed this so that `write` returns `Result<(), String>` and a `?` can be used on this line.